src/vs/base/parts/ipc/common/ipc.ts

```diff
@@ -635,8 +635,7 @@ export class ChannelClient implements IChannelClient, IDisposable {
 				e(new CancellationError());
 			};
 
-			const cancellationTokenListener = cancellationToken.onCancellationRequested(cancel);
-			disposable = combinedDisposable(toDisposable(cancel), cancellationTokenListener);
```

Comment:
I think we do still want to cancel the request and reject the promise when the disposable is disposed, but only if it the promise already been resolved. I.e. it should so as it was doing if the client is disposed with ongoing requests (otherwise those requests just hang indefinitely) but not from the `result.finally` block.

Reply:
You're right. 
I made some changes, could you take another look?